sdk/js/src/token_bridge/transfer.ts

```diff
@@ -234,6 +237,94 @@ export async function transferFromTerra(
       ];
 }
 
+/**
+ * Creates the necessary messages to transfer an asset
+ * @param walletAddress Address of the Inj wallet
+ * @param tokenBridgeAddress Address of the token bridge contract
+ * @param tokenAddress Address of the token being transferred
+ * @param amount Amount of token to be transferred
+ * @param recipientChain Destination chain
+ * @param recipientAddress Destination wallet address
+ * @param relayerFee Relayer fee
+ * @param payload Optional payload
+ * @returns Transfer messages to be sent on chain
+ */
+export async function transferFromInjective(
+  walletAddress: string,
+  tokenBridgeAddress: string,
+  tokenAddress: string,
+  amount: string,
+  recipientChain: ChainId | ChainName,
+  recipientAddress: Uint8Array,
+  relayerFee: string = "0",
+  payload: Uint8Array | null = null
+) {
+  const recipientChainId = coalesceChainId(recipientChain);
+  const nonce = Math.round(Math.random() * 100000);
+  const isNativeAsset = isNativeCosmWasmDenom(CHAIN_ID_INJECTIVE, tokenAddress);
+  const mk_action = () =>
+    payload ? "initiate_transfer_with_payload" : "initiate_transfer";
+  const mk_initiate_transfer = (info: object) =>
+    payload
+      ? {
+          asset: {
+            amount,
+            info,
+          },
+          recipient_chain: recipientChainId,
+          recipient: Buffer.from(recipientAddress).toString("base64"),
+          fee: relayerFee,
+          nonce: nonce,
+          payload: payload,
+        }
+      : {
+          asset: {
+            amount,
+            info,
+          },
+          recipient_chain: recipientChainId,
+          recipient: Buffer.from(recipientAddress).toString("base64"),
+          fee: relayerFee,
+          nonce: nonce,
+        };
+  return isNativeAsset
+    ? [
+        MsgExecuteContractInjective.fromJSON({
+          contractAddress: tokenBridgeAddress,
+          sender: walletAddress,
+          msg: {},
+          action: "deposit_tokens",
+          amount: { denom: "inj", amount: amount },
```

Comment:
do we want to hardcode "inj" here? the transfer below is for `tokenAddress`

Reply:
Fixed